modules/public/public.go

```diff
@@ -98,76 +57,51 @@ func parseAcceptEncoding(val string) map[string]bool {
 	return types
 }
 
-func (opts *Options) handle(w http.ResponseWriter, req *http.Request, opt *Options) bool {
+func (opts *Options) handle(w http.ResponseWriter, req *http.Request, fs http.FileSystem, prefix string) bool {
 	if req.Method != "GET" && req.Method != "HEAD" {
 		return false
 	}
 
 	file := req.URL.Path
 	// if we have a prefix, filter requests by stripping the prefix
-	if opt.Prefix != "" {
-		if !strings.HasPrefix(file, opt.Prefix) {
+	if prefix != "" {
+		if !strings.HasPrefix(file, prefix) {
 			return false
 		}
-		file = file[len(opt.Prefix):]
+		file = file[len(prefix):]
 		if file != "" && file[0] != '/' {
 			return false
 		}
 	}
 
-	f, err := opt.FileSystem.Open(file)
+	f, err := fs.Open(file)
```

Comment:
It may be helpful to add a comment above this stating that fs.Open does a `path.Clean("/"+...)` internally or, explicitly do it ourselves, e.g. make line 71

```go
		file = path.Clean("/" + file[len(prefix):])[1:]
```

Reply:
This will require go1.16